.drone.yml

```diff
@@ -1,121 +1,209 @@
+  when:
+    event:
+    - tag
 
-  docker-publish-agent:
-    image: plugins/docker
+- name: docker-publish-agent
+  pull: default
+  image: plugins/docker
+  settings:
+    build_args:
+    - "VERSION=${DRONE_TAG}"
+    context: package/
+    custom_dns: 1.1.1.1
     dockerfile: package/Dockerfile.agent
+    password:
+      from_secret: docker_password
     repo: rancher/rancher-agent
-    context: package/
-    tag: ${DRONE_TAG}
-    secrets: [docker_username, docker_password]
-    build_args:
-    - VERSION=${DRONE_TAG}
-    when:
-      branch: [master, release/*]
-      event: tag
+    tag: "${DRONE_TAG}"
+    username:
+      from_secret: docker_username
+  when:
+    event:
+    - tag
 
-  github_binary_prerelease:
-    image: plugins/github-release
-    prerelease: true
-    files:
-     - bin/rancher-*
+- name: github_binary_prerelease
+  pull: default
+  image: plugins/github-release
+  settings:
+    api_key:
+      from_secret: github_token
     checksum:
-     - sha256
-    secrets: [github_token]
-    when:
-      branch: [master, release/*]
-      event: tag
-      ref:
-        include: [ refs/tags/*rc* ]
-
-  github_binary_release:
-    image: plugins/github-release
+    - sha256
     files:
-     - bin/rancher-*
+    - "bin/rancher-*"
+    prerelease: true
+  when:
+    event:
+    - tag
+    ref:
+      include:
+      - "refs/tags/*rc*"
+      - "refs/tags/*alpha*"
+
+- name: github_binary_release
+  pull: default
+  image: plugins/github-release
+  settings:
+    api_key:
+      from_secret: github_token
     checksum:
-     - sha256
-    secrets: [github_token]
-    when:
-      branch: [master, release/*]
-      event: tag
-      ref:
-        exclude: [ refs/tags/*rc* ]
-
-  chart-promote:
-    image: rancher/dapper:1.11.2
-    commands:
-      - dapper chart/copy
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
-    when:
-      event: deployment
-      environment: promote-stable
-
-  chart-publish:
-    image: plugins/gcs
+    - sha256
+    files:
+    - "bin/rancher-*"
+  when:
+    event:
+    - tag
+    ref:
+      include:
+      - "refs/tags/v2*"
+      exclude:
+      - "refs/tags/*rc*"
+      - "refs/tags/*alpha*"
+
+- name: chart-publish
+  pull: default
+  image: plugins/gcs
+  settings:
+    acl:
+    - allUsers:READER
+    cache_control: "public,no-cache,proxy-revalidate"
     source: bin/chart
     target: releases.rancher.com/server-charts
-    acl:
-      - allUsers:READER
-    cache_control: public,no-cache,proxy-revalidate
-    secrets:
-      - source: google_auth_key
-        target: GOOGLE_CREDENTIALS
-    when:
-      event: [ tag, deployment ]
+    token:
+      from_secret: google_auth_key
+  when:
+    event:
+    - tag
+
+volumes:
+- name: docker
+  host:
+    path: /var/run/docker.sock
+
+trigger:
+  event:
+    exclude:
+    - promote
+---
+kind: pipeline
+name: publish
+
+platform:
+  os: linux
+  arch: amd64 
+
+steps:
+- name: chart-promote
```

Comment:
I think this step should just be in the main pipeline, or we need a copy of chart-publish here to do the chart push.